root/oauth2/Authorize.js

```diff
@@ -0,0 +1,75 @@
+/*
+ * @flow
+ * Copyright (C) 2019 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import * as React from 'react';
+
+import {
+  ACCESS_SCOPE_PROFILE,
+  ACCESS_SCOPE_EMAIL,
+  ACCESS_SCOPE_TAG,
+  ACCESS_SCOPE_RATING,
+  ACCESS_SCOPE_COLLECTION,
+  ACCESS_SCOPE_SUBMIT_ISRC,
+  ACCESS_SCOPE_SUBMIT_BARCODE,
+} from '../constants';
+import {withCatalystContext} from '../context';
+import Layout from '../layout';
+import {l, N_l} from '../static/scripts/common/i18n';
+
+const ACCESS_SCOPE_PERMISSIONS = {
+  [ACCESS_SCOPE_COLLECTION]: N_l('View and modify your private collections'),
+  [ACCESS_SCOPE_EMAIL]: N_l('View your email address'),
+  [ACCESS_SCOPE_PROFILE]: N_l('View your public account information'),
+  [ACCESS_SCOPE_RATING]: N_l('View and modify your private ratings'),
+  [ACCESS_SCOPE_SUBMIT_BARCODE]: N_l('Submit new barcodes to the database'),
+  [ACCESS_SCOPE_SUBMIT_ISRC]: N_l('Submit new ISRCs to the database'),
+  [ACCESS_SCOPE_TAG]: N_l('View and modify your private tags'),
```

Comment:
Probably stupid question, but: why can't this just be l()?

Reply:
Because these statements are executed when the file is required and loaded, not when the component renders. `l()` return a translated string for the current locale, but the locale when the worker process starts is probably English, so `l()` here would just always return an English string.

(In case it's unclear, modules aren't reloaded every time you require() them within the same process - they're executed once and then cached, so if you require() the same file again it return the same reference back.)

Reply:
I'll point out though that if you have `DEVELOPMENT_SERVER` enabled, we [explicitly clear the module cache](https://github.com/metabrainz/musicbrainz-server/blob/a6d9d1ed9aaca7999a8fc61b62cb52917deb9c37/root/server/createServer.js#L68) after every request, so the module *is* reloaded and evaluated every time you load the page in that case. :) (So you wouldn't observe any translation bug, but it would happen in production.)